File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/di/component/ApiComponent.java
package com.kalashnyk.denys.samplemvvmapp.di.component;

import com.kalashnyk.denys.samplemvvmapp.di.module.ApiModule;
import com.kalashnyk.denys.samplemvvmapp.di.scope.ApiScope;
import com.kalashnyk.denys.samplemvvmapp.repository.server.ServerCommunicator;

import dagger.Component;

@ApiScope
@Component(modules = {ApiModule.class})
public interface ApiComponent {
    ServerCommunicator getServerCommunicator();
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/di/component/ApiComponent.kt
package com.kalashnyk.denys.kotlinsample.di.component

import com.kalashnyk.denys.kotlinsample.di.module.ApiModule
import com.kalashnyk.denys.kotlinsample.di.scope.ApiScope
import com.kalashnyk.denys.kotlinsample.repository.server.ServerCommunicator
import dagger.Component

@ApiScope
@Component(modules = [ApiModule::class])
interface ApiComponent {
    val serverCommunicator: ServerCommunicator
}
